core/foundation/src/StringUtils.cxx

```diff
@@ -37,4 +41,57 @@ std::vector<std::string> Split(std::string_view str, std::string_view delims, bo
    return out;
 }
 
+/**
+ * \brief Convert (round) a value and its uncertainty to string using one or two significant digits of the error
+ * \param error the error. If the error is negative or zero, only the value is returned with no specific rounding applied, using std::to_string
+ * \param cutoff should lay between 0 and 9. If first significant digit of error starts with value <= cutoff, use two significant digits instead of two for
+ * rounding. Set this value to zero to always use a single digit; set this value to 9 to always use two digits.
+ * \param delim delimiter between value and error printed into returned string, leave default for using ROOT's latex mode
+ * \return a string with printed rounded value and error separated by "+/-" in ROOT latex mode
+ * \note The return format is `A+-B` using ios::fixed with the proper precision;
+ * for very large or very small values of the error, the format is changed from `A+-B` to (A'+-B')*1eX, with X being multiple of 3, respecting the corresponding precision.
+ * \see https://www.bipm.org/en/doi/10.59161/jcgm100-2008e, https://physics.nist.gov/cuu/Uncertainty/
```

Comment:
In the context of giving proper attribution, can you confirm the source of the code (your own, a derivation of the code proposed at https://its.cern.ch/jira/browse/ROOT-11017, somewhere else)?  Thanks.

Reply:
Sure, I wrote the code myself from scratch, I didn't really look at the example on JIRA since it used another paradigm specifying an arbitrary precision. Instead, I just followed the rule that errors have to be rounded with either one or two significant digits, not more.